src/app/pages/storage/snapshots/snapshot-table/snapshot-list.component.ts

```diff
@@ -0,0 +1,327 @@
+    private translate: TranslateService,
+    private modalService: ModalService,
+    private cdr: ChangeDetectorRef,
+    private core: CoreService,
+    private loader: AppLoaderService,
+    private matDialog: MatDialog,
+    private store$: Store<SnapshotSlice>,
+  ) {
+    if (window.localStorage.getItem('snapshotXtraCols') === 'true') {
+      this.showExtraColumns$.next(true);
+    }
+  }
+
+  ngOnInit(): void {
+    this.setupToolbar();
+    this.getSnapshots();
+  }
+
+  ngAfterViewInit(): void {
+    this.store$.dispatch(loadSnapshots({ extra: this.showExtraColumns$.value }));
+    this.modalService.onClose$.pipe(untilDestroyed(this)).subscribe(() => {
+      this.getSnapshots();
+    });
+  }
+
+  getSnapshots(): void {
+    this.store$.select(selectSnapshots).pipe(
+      map((snapshots: ZfsSnapshot[]) => {
+        const snapshotListRows = [];
+
+        for (const row of snapshots) {
+          const [datasetName, snapshotName] = row.name.split('@');
+
+          const transformedRow = {
+            id: row.name,
+            dataset: datasetName,
+            snapshot: snapshotName,
+            properties: row.properties,
+            name: row.name,
+          } as SnapshotListRow;
+
+          if (row.properties) {
+            snapshotListRows.push({
+              ...transformedRow,
+              used: parseInt(row.properties.used.rawvalue),
+              created: row.properties.creation.parsed.$date,
+              referenced: parseInt(row.properties.referenced.rawvalue),
+            });
+          } else {
+            snapshotListRows.push(transformedRow);
+          }
+        }
+
+        return snapshotListRows;
+      }),
+      untilDestroyed(this),
+    ).subscribe((snapshotListRows) => {
+      this.createDataSource(snapshotListRows);
+      this.cdr.markForCheck();
+    }, () => {
+      this.createDataSource();
+      this.cdr.markForCheck();
+    });
+  }
+
+  getConfirmOptions(): ConfirmOptions {
+    if (this.showExtraColumns$.value) {
+      return {
+        title: this.translate.instant(helptext.extra_cols.title_hide),
+        message: this.translate.instant(helptext.extra_cols.message_hide),
+        buttonMsg: this.translate.instant(helptext.extra_cols.button_hide),
+        hideCheckBox: true,
+      };
+    }
+
+    return {
+      title: this.translate.instant(helptext.extra_cols.title_show),
+      message: this.translate.instant(helptext.extra_cols.message_show),
+      buttonMsg: this.translate.instant(helptext.extra_cols.button_show),
+      hideCheckBox: true,
+    };
+  }
+
+  createDataSource(snapshots: SnapshotListRow[] = []): void {
+    this.dataSource = new MatTableDataSource(snapshots);
+    this.dataSource.sort = this.sort;
+  }
+
+  expandRow(row: SnapshotListRow): void {
+    this.expandedRow = this.expandedRow === row ? null : row;
+    this.cdr.markForCheck();
+  }
+
+  toggleExtraColumnsDialog(): void {
+    this.showExtraColumns$.next(!this.showExtraColumns$.value);
```

Comment:
This should dispatch showExtraColumnsPressed() and this should do processing / trigger the reload.